src/xrc-tests/src/tests/get_icp_xdr_rate.rs

```diff
@@ -0,0 +1,156 @@
+            set2_timestamp,
+            |exchange| match exchange {
+                xrc::Exchange::Binance(_) => Some("4.29"),
+                xrc::Exchange::Coinbase(_) => Some("4.30"),
+                xrc::Exchange::KuCoin(_) => Some("4.30"),
+                xrc::Exchange::Okx(_) => Some("4.28"),
+                xrc::Exchange::GateIo(_) => Some("4.28"),
+                xrc::Exchange::Mexc(_) => Some("4.291"),
+                xrc::Exchange::Poloniex(_) => Some("4.38"),
+            },
+        ))
+        .chain(mock_responses::exchanges::build_responses(
+            "ICP".to_string(),
+            set3_timestamp,
+            |exchange| match exchange {
+                xrc::Exchange::Binance(_) => Some("5.17"),
+                xrc::Exchange::Coinbase(_) => Some("5.18"),
+                xrc::Exchange::KuCoin(_) => Some("5.18"),
+                xrc::Exchange::Okx(_) => Some("5.16"),
+                xrc::Exchange::GateIo(_) => Some("5.16"),
+                xrc::Exchange::Mexc(_) => Some("5.171"),
+                xrc::Exchange::Poloniex(_) => Some("5.26"),
+            },
+        ))
+        .chain(mock_responses::stablecoin::build_responses(set1_timestamp))
+        .chain(mock_responses::stablecoin::build_responses(set2_timestamp))
+        .chain(mock_responses::stablecoin::build_responses(set3_timestamp))
+        .chain(mock_responses::forex::build_responses(now))
+        .collect::<Vec<_>>();
+
+    let container = Container::builder()
+        .name("get_icp_xdr_rate")
+        .exchange_responses(responses)
+        .build();
+
+    run_scenario(container, |container: &Container| {
+        let request = GetExchangeRateRequest {
+            base_asset: Asset {
+                symbol: "ICP".to_string(),
+                class: AssetClass::Cryptocurrency,
+            },
+            quote_asset: Asset {
+                symbol: "CXDR".to_string(),
+                class: AssetClass::FiatCurrency,
+            },
+            timestamp: Some(set1_timestamp),
+        };
+
+        let exchange_rate_result = container
+            .call_canister::<_, GetExchangeRateResult>("get_exchange_rate", &request)
+            .expect("Failed to call canister for rates");
+        let exchange_rate =
+            exchange_rate_result.expect("Failed to retrieve an exchange rate from the canister.");
+        assert_eq!(exchange_rate.base_asset, request.base_asset);
+        assert_eq!(exchange_rate.quote_asset, request.quote_asset);
+        assert_eq!(exchange_rate.timestamp, set1_timestamp);
+        assert_eq!(exchange_rate.metadata.base_asset_num_queried_sources, 7);
+        assert_eq!(exchange_rate.metadata.base_asset_num_received_rates, 7);
+        assert_eq!(exchange_rate.metadata.quote_asset_num_queried_sources, 11);
+        assert_eq!(exchange_rate.metadata.quote_asset_num_received_rates, 11);
+        assert_eq!(exchange_rate.metadata.standard_deviation, 81_973_860);
+        assert_eq!(exchange_rate.rate, 2_946_951_476);
```

Comment:
How do I know that these values are expected? Is it possible to add a comment that explains how the numbers are derived from the rates in the XRC?